unit_test/test_acknowledgements.py

```diff
@@ -0,0 +1,275 @@
+            output_file_path.touch()
+            language = "fil-PH"
+            volunteers = []
+            logging = "off"
+            result = nttt.acknowledgements.add_volunteer_acknowledgement(
+                csv_file_path, output_file_path, language, volunteers, logging)
+            self.assertFalse(result)
+
+    def test_no_volunteer_names(self):
+        '''
+        Test case for the situation where the list of volunteer names is empty.
+        '''
+
+        with TemporaryDirectory() as temp_folder:
+            data_folder = Path(os.getcwd(), "unit_test", "data")
+            self.assertTrue(
+                data_folder.is_dir(),
+                "Subdirectory data of directory unit_test is missing.")
+            csv_file_path = Path(data_folder, "volunteer_acknowledgements.csv")
+            output_file_path = Path(temp_folder, "final_step.md")
+            output_file_path.touch()
+            language = "nl-NL"
+            volunteers = []
+            logging = "off"
+            result = nttt.acknowledgements.add_volunteer_acknowledgement(
+                csv_file_path, output_file_path, language, volunteers, logging)
+            self.assertTrue(result)
+
+            # Check contents of final_step.md.
+            contents = output_file_path.read_text()
+            lines = contents.splitlines()
+            self.assertEqual(11, len(lines))
+            self.assertEqual("", lines[0])
+            self.assertEqual("***", lines[1])
+            self.assertEqual(self.FIRST_LINE, lines[2])
+            self.assertEqual("", lines[3])
+            self.assertEqual("[name]", lines[4])
+            self.assertEqual("", lines[5])
+            self.assertEqual("[name]", lines[6])
+            self.assertEqual("", lines[7])
+            self.assertEqual("[name]", lines[8])
+            self.assertEqual("", lines[9])
+            self.assertEqual(self.LAST_LINE, lines[10])
+
+    def test_one_volunteer_name(self):
+        '''
+        Test case for the situation where there is one volunteer name.
+        '''
+
+        with TemporaryDirectory() as temp_folder:
+            data_folder = Path(os.getcwd(), "unit_test", "data")
+            self.assertTrue(
+                data_folder.is_dir(),
+                "Subdirectory data of directory unit_test is missing.")
+            csv_file_path = Path(data_folder, "volunteer_acknowledgements.csv")
+            output_file_path = Path(temp_folder, "final_step.md")
+            output_file_path.touch()
+            language = "nl-NL"
+            volunteers = ["Volunteer One"]
+            logging = "off"
+            result = nttt.acknowledgements.add_volunteer_acknowledgement(
+                csv_file_path, output_file_path, language, volunteers, logging)
+            self.assertTrue(result)
+
+            # Check contents of final_step.md.
+            contents = output_file_path.read_text()
+            lines = contents.splitlines()
+            self.assertEqual(7, len(lines))
```

Comment:
This line-by-line content comparison is not very readable. Why not compare result strings directly? Python has multiline strings as well as handy ways to concatenate strings. See `TestEol` for some examples

Reply:
Agreed. Now str.join is my magic tool ;-)